app/tests/src/test_format.py

```diff
@@ -1,22 +1,34 @@
+import re
+
 from sqlalchemy import delete
 
-from src.db.models.document import Document
+from src.db.models.document import Chunk, ChunkWithScore, Document
 from src.format import format_guru_cards
 from src.retrieve import retrieve_with_scores
 from tests.src.test_retrieve import _create_chunks
 
 
-def _get_chunks_with_scores():
+def _get_chunks_with_scores(user_config):
     _create_chunks()
-    return retrieve_with_scores("Very tiny words.", k=2)
+    return retrieve_with_scores("Very tiny words.", user_config)
+
+
+def _unique_accordion_ids(html):
+    return set(
+        [html[m.start() + 4 : m.end() - 1] for m in re.finditer(" id=accordion-\\d*>", html)]
+    )
 
 
-def test_format_guru_cards_with_score(app_config, db_session, enable_factory_create):
+def test_format_guru_cards_with_score(monkeypatch, app_config, db_session, enable_factory_create):
     db_session.execute(delete(Document))
 
-    chunks_with_scores = _get_chunks_with_scores()
-    html = format_guru_cards(chunks_with_scores)
-    assert "accordion-1" in html
+    monkeypatch.setenv("DOCS_SHOWN_MIN_SCORE", "0.0")
+    app_config.reinit()
+    user_config = app_config.create_user_config(retrieval_k=2)
```

Comment:
Since we now have the `UserConfig` class, if we don't want to add the `DynamicAppConfig` class, we can use `app_config.create_user_config(retrieval_k=2)` or `user_config.retrieval_k=2` to override *user* configuration settings for unit tests. Note that *non-user* configuration settings in AppConfig would not be reconfigurable without `DynamicAppConfig`.